### PR TITLE
glide: use jessevdk/go-flags for consistency.

### LINKED ISSUE #804: Reference upstream repo and specific version of 'go-flags'
Currently dcrwallet references `github.com/jessevdk/go-flags` while dcrd references `github.com/btcsuite/go-flags` in `glide.yaml`. 

For consistency both should reference the same repository. Because `jessevdk/go-flags` is the upstream repo, this would be the preferred one to reference. 

Also `glide.yaml` should reference a specific version/tag instead of the tip of the default branch as is done now.

### PR DESCRIPTION
**Pull request by dnldd**
Fixes #804

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -5,11 +5,15 @@ import:
   subpackages:
   - edwards25519
 - package: github.com/btcsuite/btclog
-- package: github.com/btcsuite/go-flags
+  version: 84c8d2346e9fc8c7b947e243b9c24e6df9fd206a
+- package: github.com/jessevdk/go-flags
+  version: ^1.3.0
 - package: github.com/btcsuite/go-socks
+  version: 4720035b7bfd2a9bb130b1c184f8bbe41b6f0d0f
```
Comment (davecgh) [inline glide.yaml]: Specific commit versions should not be in the yaml.  That is what the lock file is for.

File: glide.yaml [hunk 2]
```diff
@@ -5,11 +5,15 @@ import:
...
   subpackages:
   - socks
 - package: github.com/btcsuite/goleveldb
+  version: 7834afc9e8cd15233b6c3d97e12674a31ca24602
```
Comment (davecgh) [inline glide.yaml]: Same.

File: glide.yaml [hunk 3]
```diff
@@ -19,26 +23,34 @@ import:
   - leveldb/opt
   - leveldb/util
 - package: github.com/btcsuite/websocket
+  version: 31079b6807923eb23992c421b114992b95131b55
```
Comment (davecgh) [inline glide.yaml]: Same.

File: glide.yaml [hunk 4]
```diff
@@ -19,26 +23,34 @@ import:
...
 - package: github.com/btcsuite/winsvc
+  version: f8fb11f83f7e860e3769a08e6811d1b399a43722
```
Comment (davecgh) [inline glide.yaml]: Same.

File: glide.yaml [hunk 5]
```diff
@@ -19,26 +23,34 @@ import:
...
   subpackages:
   - eventlog
   - mgr
   - svc
 - package: github.com/davecgh/go-spew
+  version: adab96458c51a58dc1783b3335dcce5461522e75
```
Comment (davecgh) [inline glide.yaml]: Same

File: glide.yaml [hunk 6]
```diff
@@ -19,26 +23,34 @@ import:
...
   subpackages:
   - spew
 - package: github.com/dchest/blake256
   version: dee3fe6eb0e98dc774a94fc231f85baf7c29d360
 - package: github.com/decred/bitset
+  version: 484b833245d5f9046e2893a6bd2e54b1df3a53a4
```
Comment (davecgh) [inline glide.yaml]: Same.

File: glide.yaml [hunk 7]
```diff
@@ -19,26 +23,34 @@ import:
...
 - package: github.com/decred/dcrrpcclient
+  version: abc326aeb97b299c7eb6b13b81353fc375e9b95d
```
Comment (davecgh) [inline glide.yaml]: Same.

File: glide.yaml [hunk 8]
```diff
@@ -19,26 +23,34 @@ import:
...
 - package: github.com/decred/dcrutil
+  version: 4ce0f41cf3f957fcdf3f0de049602ac834449b9c
```
Comment (davecgh) [inline glide.yaml]: Same.

File: glide.yaml [hunk 9]
```diff
@@ -19,26 +23,34 @@ import:
...
   subpackages:
   - bloom
 - package: golang.org/x/crypto
+  version: 6914964337150723782436d56b3f21610a74ce7b
```
Comment (davecgh) [inline glide.yaml]: Same.

File: glide.yaml [hunk 10]
```diff
@@ -19,26 +23,34 @@ import:
...
   subpackages:
   - ripemd160
   - ssh/terminal
 - package: github.com/jrick/logrotate
+  version: a93b200c26cbae3bb09dd0dc2c7c7fe1468a034a
```
Comment (davecgh) [inline glide.yaml]: Same.

**Comment by dnldd** [comment]
@davecgh the repos for the highlighted lines for the dependencies above don't have numbered versions. They either have outdated releases (in comparison to the versions specified in the `.lock` file) or none at all. The only way to specify dep versions is via commits (as done above), without this there isn't any way to specify versions in the `.yaml` file. 

Because the dep versions aren't set in dcrd's current `.yaml` file, running `glide up` updates dependencies to their respective repo branch tips (which is undesirable). Please reconsider.

**Comment by davecgh** [comment]
Running `glide up` is _supposed_ to update the lock file to latest versions.  It's up to the maintainers (aka us) to choose whether or not we want to update the project dependencies to those versions or not.  This change would completely break the intended purpose of `glide up`.

**Comment by jrick** [comment]
I'm going to link to the cargo documentation here because i think it explains it better than glide, but glide follows the exact same methodology.

http://doc.crates.io/guide.html#cargotoml-vs-cargolock

We should not be specifying exact commits in glide.yaml except under very particular circumstances.

**Comment by dnldd** [comment]
@jrick thanks for that link. @davecgh reverting.

**Comment by dnldd** [comment]
@davecgh updated the PR, should be good now.

File: glide.yaml [hunk 11]
```diff
@@ -1,11 +1,11 @@
...
 import:
 - package: github.com/agl/ed25519
-  version: 278e1ec8e8a6e017cd07577924d6766039146ced
   subpackages:
   - edwards25519
 - package: github.com/btcsuite/btclog
-- package: github.com/btcsuite/go-flags
+- package: github.com/jessevdk/go-flags
+  version: ^1.3.0
 - package: github.com/btcsuite/go-socks
```
Comment (jrick) [inline glide.yaml]: the btcsuite fork of go-flags should be removed since we are switching to upstream.

Comment (jrick) [reply glide.yaml]: commented on wrong file, it might be a transient dep in case we need to figure out what imported package is requiring it.

**Comment by davecgh** [comment]
Need a rebase to resolve merge conflicts, but otherwise this looks ready.

**Pull request merged**